src/runtime_src/core/common/query_requests.h

```diff
@@ -725,6 +726,32 @@ struct mem_topology_raw : request
   get(const device*) const = 0;
 };
 
+struct mem_topology : request
+{
+  struct memory_data {
+    std::string xclbin_uuid;
+    uint32_t hw_context_slot;
+    uint8_t m_type;          // enum corresponding to mem_type.
+    uint8_t m_used;          // if 0 this bank is not present
+    union {
+        uint64_t m_size;     // if mem_type DDR, then size in KB;
+        uint64_t route_id;   // if streaming then "route_id"
+    };
+    union {
+        uint64_t m_base_address; // if DDR then the base address;
+        uint64_t flow_id;        // if streaming then "flow id"
+    };
+    unsigned char m_tag[16]; // DDR: BANK0,1,2,3, has to be null terminated; if streaming then stream0, 1 etc
```

Comment:
I don't like this.  It's a copy from xclbin.h and I don't understand why?

Reply:
Understood. I will move the formatting back into info_memory and remove the new query requests. We should not be doing any formatting in the query requests.